src/restic/node.go

```diff
@@ -493,12 +559,35 @@ func (node *Node) fillExtra(path string, fi os.FileInfo) error {
 	case "fifo":
 	case "socket":
 	default:
-		err = errors.Errorf("invalid node type %q", node.Type)
+		return errors.Errorf("invalid node type %q", node.Type)
+	}
+
+	if err = node.fillExtendedAttributes(path); err != nil {
+		return err
 	}
 
 	return err
 }
 
+func (node *Node) fillExtendedAttributes(path string) error {
+	if node.Type == "symlink" && node.LinkTarget == "invalid" {
```

Comment:
I've deliberately included a symlink node with a non-existing target in the tests for the `RestoreAt()` function. restic should be able to restore even those symlinks which do not point to a valid target.

Can symlinks have their own extended attributes? According to the tests that I just did locally (on Linux), they cannot. So restic does not need to save extended attributes for symlinks, and it must not try to restore the extended attributes for them.

I think that's a bug in the code you contributed. What do you think?